### PR TITLE
Feature/different game states

### PR DESCRIPTION
**Pull request by JR-Mitchell**
### What?
- Sets up 3 game states:
    - Waiting for players
    - Playing round
    - Waiting for next round
- Sets up event listening for the `roundOver` event

### Why?
UI feedback for the gameplay loop

### CODE REVIEW

File: client/src/data/roundOver.d.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+type RoundOver = {
+    "winningTeam": number,
+    "winningWord": string,
+    "team1score": number,
+    "team2score": number
+}
```
Comment (fjebaker) [inline client/src/data/roundOver.d.ts]: Exporten this!

File: client/src/pages/MainPage.tsx [hunk 2]
```diff
@@ -61,16 +66,18 @@ const MainPage = () => {
             events.keyPress(event);
             break;
           case "setup":
-            events.setup(event, setWordLengths, setOwnAnswers, setEnemyAnswers);
+            events.setup(event, setGameState, setWordLengths, setOwnAnswers, setEnemyAnswers);
             break;
+          case "roundOver":
+            events.roundOver(event, user_data ? user_data.team : -1, setGameState, setWinState);
```
Comment (fjebaker) [inline client/src/pages/MainPage.tsx:72]: `HALT;`

**Review by fjebaker — CHANGES_REQUESTED** [review]

**Review by RiFactor — APPROVED** [review]

**Review by Shellywell123 — APPROVED** [review]
you do you

**Review by fjebaker — APPROVED** [review]

**Comment by Shellywell123** [comment]
just here for the memes

**Pull request merged**